Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/ContestsPage.tsx

```diff
@@ -164,31 +172,68 @@ const ContestsPage = () => {
         [ contests, currentPage, handlePageChange, isLoaded, pagesInfo, renderContest ],
     );
 
+    const updateBreadcrumbAndNavigateToCategory = useCallback(
+        (breadcrumb: ICategoriesBreadcrumbItem) => {
+            const category = categoriesFlat.find(({ id }) => id.toString() === breadcrumb.id.toString());
+
+            updateBreadcrumb(category, categoriesFlat);
+            navigate(getContestCategoryBreadcrumbItemPath(breadcrumb.id));
+        },
+        [ categoriesFlat, navigate, updateBreadcrumb ],
+    );
+
+    const renderCategoriesBreadcrumbItem = useCallback(
+        (categoryBreadcrumbItem: ICategoriesBreadcrumbItem) => {
+            const { value, isLast } = categoryBreadcrumbItem;
+            const classNames = concatClassNames(styles.breadcrumbBtn, isLast
+                ? styles.breadcrumbBtnLast
+                : '');
+
+            return (
+                <Button
+                  type={ButtonType.plain}
+                  className={classNames}
+                  onClick={() => updateBreadcrumbAndNavigateToCategory(categoryBreadcrumbItem)}
+                  text={value}
+                />
+            );
+        },
+        [ updateBreadcrumbAndNavigateToCategory ],
+    );
```

Comment:
There is almost the same function in the Contest.tsx . Can't it be reused and controlled by parameters ?